README.md

```diff
@@ -9,21 +9,72 @@ rustup target add riscv64gc-unknown-none-elf
 cargo build
 ```
 
-## running
+## Running
 
-There is a `Makefile` provided with targets for running various hosts in qemu.
+There is a `Makefile` provided with targets for running various hosts in QEMU.
 
 ### Prerequisites
 
-- rust toolchain from [rustup](rustup.rs)
-- rustup target add riscv64gc-unknown-none-elf
-- install `gcc-riscv64-unknown-elf` for `riscv64-unknown-elf-objcopy`
-- qemu-system-riscv64 - build using  qemu
-  [instructions](https://wiki.qemu.org/Hosts/Linux) with
+Toolchains:
+- Rust: Install [rustup](rustup.rs), then: `rustup target add riscv64gc-unknown-none-elf`
+- GCC: Install `gcc-riscv64-unknown-elf`
+
+QEMU:
+- Out-of-tree patches are required; see table below.
+- Build using QEMU [instructions](https://wiki.qemu.org/Hosts/Linux) with
   `--target-list=riscv64-softmmu`
-  - Note qemu 7.0.0 has a bug emulating riscv and a newer version from git must be used.
 
-## Test VM
+OpenSBI:
+- Use source from upstream: https://github.com/riscv-software-src/opensbi
+- Build: `PLATFORM=generic CROSS_COMPILE=riscv64-unknown-linux-gnu- make`
```

Comment:
Is OpenSBI from source still needed? I've bene using the qemu default installed firmware.

Reply:
Default OpenSBI in Qemu works for me as well.

Reply:
Works for me as well (modulo the FDT issue). Removed.